encryption/client/src/main/java/org/wildfly/security/encryption/client/EncryptedExpressionResolver.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2024 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.encryption.client;
+
+import org.wildfly.security.credential.SecretKeyCredential;
+import org.wildfly.security.credential.store.CredentialStore;
+import org.wildfly.security.credential.store.CredentialStoreException;
+import org.wildfly.security.encryption.client._private.ElytronMessages;
+
+import javax.crypto.SecretKey;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+import java.util.Map;
+
+import static org.wildfly.common.Assert.checkNotNullParam;
+import static org.wildfly.security.encryption.CipherUtil.decrypt;
+import static org.wildfly.security.encryption.CipherUtil.encrypt;
+
+/**
+ * A class used to resolve encrypted expressions using secret key within credential stores.
+ * Contains a collection of multiple resolver configurations.
+ *
+ * @author <a href="mailto:prpaul@redhat.com">Prarthona Paul</a>
+ */
+
+public class EncryptedExpressionResolver {
+    private volatile String prefix;
+    private volatile String completePrefix;
+    private volatile String defaultResolver;
+    private volatile Map<String, ResolverConfiguration> resolverConfigurations;
+
+    public EncryptedExpressionResolver() {
+    }
+
+    public String resolveExpression(String expression, EncryptionClientConfiguration config) {
+        checkNotNullParam("expression", expression);
+        checkNotNullParam("encrypted expression configuration", config);
+        return resolveExpressionInternal(expression, config);
+    }
+
+    public Map<String, ResolverConfiguration> getResolverConfiguration() {
+        return resolverConfigurations;
+    }
+
+    private String resolveExpressionInternal(String fullExpression, EncryptionClientConfiguration config) {
+        assert config != null;
+
+        if (fullExpression.length() > 3) {
+            String expression = fullExpression.substring(2, fullExpression.length() - 1);
+
+            if (expression.startsWith(completePrefix)) {
```

Comment:
If completePrefix has not been set a NullPointerException will be thrown.

Reply:
fixed! 
Thanks @rsearls